### PR TITLE
Add NeighborPackagedData to Burgers subcell

### PR DESCRIPTION
**Pull request by yoonso0-0**
## Proposed changes

- [ ] Dependent on #3673

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: src/Evolution/Systems/Burgers/Subcell/NeighborPackagedData.hpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    using derived_boundary_corrections =
+        typename std::decay_t<decltype(boundary_correction)>::creatable_classes;
+
+    // perform reconstruction
+    tmpl::for_each<derived_boundary_corrections>([&](auto
+                                                         derived_correction_v) {
+      using derived_correction =
+          tmpl::type_from<decltype(derived_correction_v)>;
+      if (typeid(boundary_correction) == typeid(derived_correction)) {
+        // U, NormalDotFlux<U>, AbsCharSpeed
```
Comment (nilsdeppe) [inline src/Evolution/Systems/Burgers/Subcell/NeighborPackagedData.hpp]: I think this comment depends on the Riemann solver

File: src/Evolution/Systems/Burgers/Subcell/NeighborPackagedData.hpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        using dg_package_field_tags =
+            typename derived_correction::dg_package_field_tags;
+        // U, Flux<U>
+        using dg_package_data_argument_tags =
+            tmpl::append<evolved_vars_tags, fluxes_tags>;
+
+        const auto& element = db::get<domain::Tags::Element<1>>(box);
+
+        // Variables to store packaged data
+        Variables<dg_package_field_tags> packaged_data{0};
```
Comment (nilsdeppe) [inline src/Evolution/Systems/Burgers/Subcell/NeighborPackagedData.hpp]: Just default construct this and on line 121 (2 lines below), no size needed

**Review by nilsdeppe — CHANGES_REQUESTED** [review]
You can squash and rebase immediately

**Comment by yoonso0-0** [comment]
Rebased and squashed. Thanks!

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**